### DISCUSSION TITLE
Symmetrization of the Dynamical Matrix for the Thermal Property Calculation

### DISCUSSION THREAD

**Discussion by ksauni** [ttadano/alamode#190] [General]
Dear All,

Can I ask a question about the construction of the dynamical matrix with the force constant matrix?

To compute the thermal transmission coefficient with non-equilibrium green function (NEGF) method, the surface green function of the lead needs to be calculated and the off-diagonal part of the dynamical matrix is required.

On the other hand, I also heard that the dynamical matrix needs to be symmetrized so that the upper and lower triangle blocks in the dynamical matrix are hermitian conjugate to each other.

Would you please tell me how to symmetrize the dynamical matrix, which is built with the force constant matrix?

Thank you in advance.

Kind regards,

Kieran

**Comment by ttadano** [comment]
Hi, 

If you set `PRINTEVEC = 1`, the anphon code will generate `PREFIX.evec` file, which containts the eigenvalues and eigenvectors of the dynamical matrices on the given k points. Please read these data and reconstruct the dynamical matrix.

The dynamical matrix is Hermite, so you don't have to symmetrize it by yourself.

**Comment by ksauni** [comment]
> Hi,
> 
> If you set `PRINTEVEC = 1`, the anphon code will generate `PREFIX.evec` file, which containts the eigenvalues and eigenvectors of the dynamical matrices on the given k points. Please read these data and reconstruct the dynamical matrix.
> 
> The dynamical matrix is Hermite, so you don't have to symmetrize it by yourself.

Dear Tadano,

Thank you for the reply.

I want to construct the dynamical matrix on a dense k point mesh. If I use the eigenvalue and eigenvector to construct the dynamical matrix on a dense k point mesh, I have to output a lot of data on different k points, which would occupy a lot of space on the hard drive.

Therefore, I want to use the force constant matrix, multiplied by a phase factor, to construct the dynamical matrix. In this way, I just need to output one set of data (one force constant matrix) and it would save a lot of space on the hard drive.

I am not sure whether the dynamical matrix is hermitian or not, if I construct it through the force constant matrix from ALAMODE code. Would you please give me more explanations?

Thank you again.

Kieran

**Comment by ksauni** [comment]
Dear Tadano,

Thank you for the reply.

I want to construct the dynamical matrix on a dense k point mesh. If I use the eigenvalue and eigenvector to construct the dynamical matrix on a dense k point mesh, I have to output a lot of data on different k points, which would occupy a lot of space on the hard drive.

Therefore, I want to use the force constant matrix, multiplied by a phase factor, to construct the dynamical matrix. In this way, I just need to output one set of data (one force constant matrix) and it would save a lot of space on the hard drive.

I am not sure whether the dynamical matrix is hermitian or not, if I construct it through the force constant matrix from ALAMODE code. Would you please give me more explanations?

Thank you again.

Kieran

________________________________

**Comment by ttadano** [comment]
In `alm`, you can save the force constant matrix by setting `HESSIAN = 1` in the `&general` field. From this information, you can construct the dynamical matrix at arbitrary k points by multiplying the phase factor and mass factor.
